### PR TITLE
Signed resolution for appointing officers

### PR DESCRIPTION
**Pull request by patcon**
Re-ticketed from https://github.com/hyphacoop/organizing/issues/99#issuecomment-531931970

This is just one way that we might choose to keep our records, assuming keybase:

### Signing

    # GnuPG
    $ gpg --sign --armor \
      --output resolution-001.md.yurkowashere.asc \
      resolution-001.md

    # keybase cli
    $ keybase pgp sign \
      --infile resolutions/resolution-001.md \
      --outfile resolutions/resolution-001.md.yurkowashere.asc

### Verifying

    # GnuPG
    $ gpg --verify resolution-001.md.yurkowashere.asc

    # keybase cli
    $ keybase pgp verify \
      --infile resolutions/resolution-001.md.yurkowashere.asc

EDIT: on yurko's input, moved from instructions for detached to attached with clearsign.

Can also now drop the `resolution-*.md.*.asc` into https://keybase.io/verify

## To Do

[Consensed on](https://github.com/hyphacoop/organizing/pull/113#issuecomment-540220549) Option A: `resolutions/resolution-001/option-A`
https://github.com/hyphacoop/organizing/tree/res/001-appoint-officers/resolutions/resolution-001/option-A

- [x] final review of [the resolution](https://github.com/hyphacoop/organizing/blob/res/001-appoint-officers/resolutions/resolution-001/option-A/resolution-001.md) for glaring issues
  - [x] @YurkoWasHere 
  - [x] @benhylau 
  - [x] @dcwalk 
  - [x] @patcon 
- [x] all directors sign
  - [x] @YurkoWasHere 
  - [x] @benhylau 
  - [x] @dcwalk 
  - [x] @patcon 
- [x] clean up folder structure back to `/resolutions/resolution-001.md`

### CODE REVIEW

**Comment by patcon** [comment]
Note that detached signatures as files are more versatile than signed commits because the legal digital signature can travel with the file (for example, if we migrate resolutions to a new repo in the future)

**Comment by YurkoWasHere** [comment]
GPG does not seem to confirm the key signed with keybase ?

```
user@laptop:~/tmp/a/organizing/resolutions$ gpg --verify resolution-001.md.yurkowashere.asc resolution-001.md
gpg: Signature made Wed 18 Sep 2019 11:25:34 AM EDT
gpg:                using RSA key CA9A1680A60A2987
gpg: BAD signature from "email" [unknown]
user@laptop:~/tmp/a/organizing/resolutions$ 
```

GPG has a sigining method called clearsign w

`gpg  --output resolution-001.md.signed --clearsign resolution-001.md`

`resolution-001.md.signed` then can replace `resolution-001.md`
It also supports mutiple signers by simpy adding additional `-----BEGIN PGP SIGNATURE-----` clauses

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

<!-- Title: -->
# Ordinary resolution to appoint first officers

<sup>[From template][template]</sup>

<!-- Preamble: -->
Whereas the co-operative wants to meet legal requirements by having the minimum required officers,

<!-- Resolution: -->
Therefore be it resolved that the board of directors has appointed the following officers, effective immediately:

| Office    | Name of officeholder |
|-----------|----------------------|
| President | Yurko         |
| Secretary | Benedict Lau         |

[Signed by][sigfile] President: Yurko
Date: 2019-09-18

<!-- Links -->
   [template]: -resolution-xxx.md
   [sigfile]: resolution-001.md.yurkowashere.asc
-----BEGIN PGP SIGNATURE-----

iQIzBAEBCAAdFiEEJOsmlyFcYwgsPB4y2wKgTqpOYwwFAl2C8HcACgkQ2wKgTqpO
...
=jw3J
-----END PGP SIGNATURE-----
```

**Comment by patcon** [comment]
gpg has `--detached` too. But in favour of attached signatures: we can make super-easy instructions for verifying on keybase.io:

> Copy signed `*.asc` message into https://keybase.io/verify and click the button.

If we kept the human-readable markdown, then that seems all good :) :+1:

Will push proposed change

**Comment by patcon** [comment]
@dcwalk [on pgp/keybase for sigs](https://matrix.to/#/!wvFLvXHaNrxWWwFOKj:tomesh.net/$15687663045946BEKBI:tomesh.net?via=tomesh.net&via=matrix.org&via=elon.li):

> I don't have strong thoughts, but this seems to render a process highly technical in a way that feels at odds with other discussions about tools, knowledge and expectations

@YurkoWasHere [asked about obfuscating or removing his name](https://matrix.to/#/!JloFJtoONurMdgcZCa:tomesh.net/$15688652006943tehML:tomesh.net?via=tomesh.net&via=elon.li).

**Comment by patcon** [comment]
Resources for next appointment:

This wizard allows creation of resolution templates for lots of standard actions in Ontario (can copy html via inspector without paying):
https://www.lawdepot.com/contracts/corporate-resolution/?s=QS&g=QGDocumentType

Reworked version of an appointment resolution where all directors sign:
https://gist.github.com/patcon/3e2c4f8f5746ba33d815b344ffa63e2f

We can leave this for next time if others don't find this urgent. (We can just use this better template next time we appoint officers)

**Comment by patcon** [comment]
To clarify, **I think we should approve and merge this**. If we want to do anything more (i.e., more signatures; improved language) we can override it later

**Comment by patcon** [comment]
Related chat context: https://matrix.to/#/!wvFLvXHaNrxWWwFOKj:tomesh.net/$15689335857286kSdOt:tomesh.net?via=tomesh.net&via=matrix.org&via=elon.li

**Comment by YurkoWasHere** [comment]
Applied commit to remove extra files in favor of embedded signatures. This seems to make the folder less confusing with multiple files.

If this is not desired feel free to revert the commit.

**Comment by benhylau** [comment]
I think embedded is easier to work with. Confirmed that keybase verified @YurkoWasHere's gpg signature :D the flow looks great. Thanks for working this out @patcon @YurkoWasHere

**Comment by patcon** [comment]
This reverted some things I'd done with intention, and also removed your last name, which I don't think we had agreement on as appropriate?

**Comment by benhylau** [comment]
This is how it looks after I also signed with `gpg --output resolution-001.md.signed --clearsign resolution-001.md`:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

- -----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

<!-- Title: -->
# Ordinary resolution to appoint first officers

<sup>[From template][template]</sup>

<!-- Preamble: -->
Whereas the co-operative wants to meet legal requirements by having the minimum required officers,

<!-- Resolution: -->
Therefore be it resolved that the board of directors has appointed the following officers, effective immediately:

| Office    | Name of officeholder |
|-----------|----------------------|
| President | Yurko J              |
| Secretary | Benedict Lau         |

Signed by President: Yurko J
Date: 2019-09-18

<!-- Links -->
   [template]: -resolution-xxx.md
- -----BEGIN PGP SIGNATURE-----

wsBcBAEBCAAQBQJdgrAxCRDKmhaApgophwAAWdoIAGiO/++vI45BefBWDjF+62bw
GXfgETrlsmbpQWHXHYtwey3QYNlO/61MYmpIlyQ3v/JGabh/2D9WGU+32Iovo5gy
DU6cgPtVc/pyFCQGR5fLTxnInMAgF4/BMHU0uEYkSxPRLL1jpHzlTxnuckV+ImXN
kYATsApzIJHlugzwxzObQ2BmvOPjn5y6TTYhSVjI8EtWZgnDHt/is6wuqq1uIqnm
amR5a8f7dmff7xy/HqV/4HW9YVUve6SiexPk9qOvGfVUob+hkfVu/8YROIeJddIi
/aWnielTSd/oKhslpozM/eYHknjyHpXJSnZQQzxN4WElxnYgv+WoDgvhS785hwU=
=43JZ
- -----END PGP SIGNATURE-----
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v1

iQEcBAEBAgAGBQJdj5RAAAoJEJ2fKFZnt5suFZkIAIipP6laICMFpT+PRySd4Zui
jiHr8Rip42f4l6RfisjazYdeIIHZdZw8HBIKWGa6jZ1+uL5PWwxlG4r3cmEj+iIV
/LJ2KiCUGBQfoUE3Pnc4inau7wZv5tXnpJo75354e8sV+Ff7EsLQyQUGAMmIx0Dh
MZ0kn3rsrDV1p4B78fvlhga9aH5z6KOgirMZtbjRxBYVtJ2m8Ukka+wMOwPn1gS/
k44Rqo6YIq3RB1H/Ol6Y5tVFQ9Ln0yD5/zCw49nu1o2QNXvuT5z4On+jBMW19KIp
0OsYdRI0h+8Ou6376Tfrt2buG2OvgcZ3hsHHX9CFH97yxoeGmXUCaVDAKUTfIyI=
=F4oC
-----END PGP SIGNATURE-----
```

It nests the full signed message with signature into a new message, which I signed. It's not exactly pretty, especially after multiple people have signed there will be many levels of nesting. So it's not simply appending more signatures to bottom as @YurkoWasHere suggested.

Verification on Keybase becoems multistep. You put the whole text in and click Verify, it will peel the first layer, then you need to Verify again to verify the next signature and so on...

I favour @patcon's detached signatures using:

```
gpg --output resolution-001.md.benhylau.sig --armor --detach-sig resolution-001.md
```

But it is unclear to me how the https://keybase.io/verify UI can verify detached signatures.

File: resolutions/-resolution-xxx.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+<!-- Resolution: -->
+Therefore be it resolved that...
+<!-- the decision or statement -->
+
+Signed by President: [president's name]
+Date: YYYY-MM-DD
+
+<!-- Links -->
+   [template]: -resolution-xxx.md
```
Comment (benhylau) [inline resolutions/-resolution-xxx.md]: Can we remove this in the filename? There is no `-` in the actual file! We only do this is we decide to put `YYYY-MM-DD` ahead of it which may not be a bad idea. Though we should probably include `YYYY-MM-DD` as part of the filename. Leading with `-` makes all the cli editors think it's a parameter.

```suggestion
   [template]: resolution-xxx.md
```

Comment (patcon) [reply resolutions/-resolution-xxx.md]: ah meant to prefix with the dash in filename as well. I'm fine either way. Thought dash was an informal convention for keeping templates at top on github

Comment (patcon) [reply resolutions/-resolution-xxx.md]: date prefix is fine by me too. though numbers should keep things in order as well, unless i'm missing some consideration

**Comment by YurkoWasHere** [comment]
GPG does support Multiple signatures

1) you need to sign the ORIGINAL doc without any amount

`keybase pgp sign --clearsign -i  resolution-001.md.orig`
 
2) You must manually append just the PGP key to the end of the file

```
-----BEGIN PGP SIGNATURE-----

wsFcBAEBCAAQBQJdj+HGCRCEw4BZBAdIFQAA5i8QAE2RFKLNtjoc6MjCJqGrtQDb
g6xEZuLnDc7p4Cr4i+oZW17adRcG6h41b5cSDd/lG+svAVSapdvb+MmwLdJv/Q82
...
2tcMXLfsOEaBtQnjOsVLweudR9pzDmGakOHwL9NMLxetIaBYrtxb5x+pBMX26/Of
eFFkshks9IpeFPrvupVn
=cuRp
-----END PGP SIGNATURE-----
-----BEGIN PGP SIGNATURE-----

wsFcBAEBCAAQBQJdj+J5CRCEw4BZBAdIFQAAJG4QAF7Xxctny7SJi2vjHFiVlFI/
nBsdn5lv7V6n41Qz90XDr8EmgZnL4A+hrcG44liKq+iNU0skFDB2xyJwPsngW7GZ
..
ZZtlwj3SESBEmxGN3RR+6rm2HEZn9rUQBcaFyfLKbsDX+LB58Rub51GzrFJYD9ys
HHlySGd9a0/JYnW8+z15
=49qK
-----END PGP SIGNATURE-----
```

3) keybase will ONLY check the first signature

```
keybase pgp verify -i resolution-001.md
Signature verified. Signed by yurkowashere 1 week ago (2019-09-18 18:31:13 -0400 EDT).
PGP Fingerprint: 1de25e804667c3edc0b6593eca9a1680a60a2987.
```

4) gpg will verify all

```
gpg: Signature made Wed 18 Sep 2019 06:31:13 PM EDT
gpg:                using RSA key CA9A1680A60A2987
gpg: Good signature from "Yurko J <yurko@mail-me.ca>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 1DE2 5E80 4667 C3ED C0B6  593E CA9A 1680 A60A 2987
gpg: Signature made Sat 28 Sep 2019 06:42:14 PM EDT
gpg:                using RSA key 84C3805904074815
gpg: Good signature from "Yurko Test <yurko@mail-me.ca>" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 2BBD D11D 60E8 2197 7807  9F7A 84C3 8059 0407 4815
```

**Comment by YurkoWasHere** [comment]
Keybase detached signatures not verified by GPG

```
user@laptop:~/code/hypha/organizing/resolutions$ keybase pgp verify -d detached -i resolution-001.md
Signature verified. Signed by yurkowashere 1 week ago (2019-09-18 18:31:13 -0400 EDT).
PGP Fingerprint: 1de25e804667c3edc0b6593eca9a1680a60a2987.
user@laptop:~/code/hypha/organizing/resolutions$ gpg --verify detached resolution-001.md
gpg: Signature made Sat 28 Sep 2019 06:49:31 PM EDT
gpg:                using RSA key 84C3805904074815
gpg: BAD signature from "Yurko Test <yurko@mail-me.ca>" [unknown]
user@laptop:~/code/hypha/organizing/resolutions$ 
```

**Comment by patcon** [comment]
> I favour @patcon's detached signatures using:
>
>     gpg --output resolution-001.md.benhylau.sig --armor --detach-sig resolution-001.md

fwiw, I favour

    gpg --output resolution-001.md.benhylau.sig --armor resolution-001.md

Also, not sure if it was noticed, but I built a functional attached signature from yurko's detached one + the original content, without him recreating it. Same is possible for the reverse: i can take apart the attached/embedded sig and create a detached one with just a text editor.

So which we use is kinda academic, as someone can modify them later (though we can't change the markdown). I suggest embedded/attached sigs only because it's just a little easier to copy-paste them into keybase. Related: https://github.com/keybase/keybase-issues/issues/2170#issuecomment-533434710

**Comment by YurkoWasHere** [comment]
> Also, not sure if it was noticed, but I built a functional attached signature from yurko's detached one + the original content, without him recreating it. 

This method can be used to stack the signatures see https://stackoverflow.com/questions/37725969/several-pgp-signatures-for-one-file

Proposed method by @patcon :+1: 

I don't like cluttering this with multiple files for the same resolution, it can get confusing.  

However at present i don't think there is a clean way of getting around that issue with multiple signatures.

> I suggest embedded/attached sigs only because it's just a little easier to copy-paste them into keybase

I agree but for a different reason - GPG not verifying when using detached from KeyBase [noted here](https://github.com/hyphacoop/organizing/pull/113#issuecomment-536231261)

tl;dr re Multiple Signatures method

Embed (clearsign) of the resolution as individual files 

 - Can be verified with GPG or KeyBase
 - Each person can commit their own signature 
 - A single file does not need to be modified by  multiple parties (conflicts!)

**Comment by benhylau** [comment]
So are we doing [this](https://gist.github.com/patcon/3e2c4f8f5746ba33d815b344ffa63e2f) and have each person commit a file that looks something like this?

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

<!-- Title: -->
# Ordinary resolution to appoint first officers

<sup>[From template][template]</sup>

<!-- Preamble: -->
Whereas the co-operative wants to meet legal requirements by having the minimum required officers,

<!-- Resolution: -->
Therefore be it resolved that the board of directors has appointed the following officers, effective immediately:

| Office    | Name of officeholder |
|-----------|----------------------|
| President | Yurko         |
| Secretary | Benedict Lau         |

[Signed by][sigfile] President: Yurko
Date: 2019-09-18

<!-- Links -->
   [template]: -resolution-xxx.md
   [sigfile]: resolution-001.md.yurkowashere.asc
-----BEGIN PGP SIGNATURE-----

iQIzBAEBCAAdFiEEJOsmlyFcYwgsPB4y2wKgTqpOYwwFAl2C8HcACgkQ2wKgTqpO
...
=jw3J
-----END PGP SIGNATURE-----
```

This way the Keybase verify will work?

**Comment by patcon** [comment]
Note: Removed "clearsign" from OP post, as we're not using it. OP post is up-to-date with signing instructions.

**Comment by patcon** [comment]
After the standup today ([notes](https://hackmd.io/@patcon/ryMxS3okB/https%3A%2F%2Fhackmd.io%2FaLSKteWoQf6VPI2eOgfpwA%3Fview?type=book)), we consensed among those present:
- to use option A with obfuscated name: https://github.com/hyphacoop/organizing/tree/f6b90c0c73fc003047b81971708601c26f77e3b7/resolutions/resolution-001/option-A
- to keep the resolutions in an open repo for now (though may decide to close in future, if others have concern with that)

Will add some tasks for the last signings, and maybe a final review (since the resolution copy has changed)

**Comment by patcon** [comment]
Yay! Done!
